python_modules/dagster/dagster/_core/definitions/asset_dep.py

```diff
@@ -54,6 +55,16 @@ def __new__(
     ):
```

Comment:
should `AssetDep` be marked experimental? It's used internally to pass around dependency info, so marking as experimental would require us to manually silence warnings when we make `AssetDeps` ourselves. very doable, but want to get agreement before adding all the silencing in